release/ray_release/config.py

```diff
@@ -47,10 +52,25 @@ def read_and_validate_release_test_collection(
 ) -> List[Test]:
     """Read and validate test collection from config file"""
     with open(config_file, "rt") as fp:
-        test_config = yaml.safe_load(fp)
+        tests = parse_test_definition(yaml.safe_load(fp))
 
-    validate_release_test_collection(test_config, schema_file=schema_file)
-    return test_config
+    validate_release_test_collection(tests, schema_file=schema_file)
+    return tests
+
+
+def parse_test_definition(test_definitions: List[TestDefinition]) -> List[Test]:
+    tests = []
+    for test_definition in test_definitions:
+        if "flavors" not in test_definition:
+            tests.append(test_definition)
+            continue
+        flavors = test_definition.pop("flavors")
+        for flavor in flavors:
+            test = copy.deepcopy(test_definition)
+            flavor["name"] = test["name"] + "-" + flavor["name"]
```

Comment:
I feel that flavor's `name` is handled differently than other fields. other fields are directly updating, but flavor's name is appending a suffix.

maybe change the key name from `name` to `SUFFIX` or `__suffix__`? and do not auto append the connecting `-`, so that the magic is clearer?

Reply:
I kind of want to have a consistent delimiter between the test name and the variant name so it is easy to read. I'm open to everything, so they could be:

test_name-variant_name
test_name@variant_name
[variant_name] test_name

What are your thoughts on this?

Reply:
hmm.. maybe use dot `.` then? will that work (is that char allowed?)

`_` and `-` are often mutually exclusive in naming conventions, because they kind of look the same.